utils/utils.go

```diff
@@ -934,6 +972,92 @@ func CustomizeLibertyFileMountXML(mountingPasswordKeySecret *corev1.Secret, moun
+			strings.HasPrefix(scriptCmd, ReadinessProbeFileBasedScriptName) {
+			probe = &corev1.Probe{}
+		}
+	}
+	return probe
+}
+
+func configureFileBasedProbeExec(instance *olv1.OpenLibertyApplication, probe *corev1.Probe, scriptName string, probeFile string) {
+	probe = getProbeWithoutHandlers(probe) // remove any preset handlers configured to this probe
+	probesConfig := instance.Spec.Probes
+	cmdList := []string{scriptName}
+	if scriptName == StartupProbeFileBasedScriptName {
+		// Set timeout seconds for the startup probe
+		timeoutSeconds := int32(1)
+		if probe.TimeoutSeconds > 0 {
+			timeoutSeconds = probe.TimeoutSeconds
+		}
+		cmdList = append(cmdList, fmt.Sprintf("-t %d", timeoutSeconds))
+	} else {
+		// Set period seconds for all other probes
+		periodSeconds := int32(10)
+		if probe.PeriodSeconds > 0 {
+			periodSeconds = probe.PeriodSeconds
+		}
+		cmdList = append(cmdList, fmt.Sprintf("-p %d", periodSeconds))
+	}
+	if probesConfig.FileDirectory != nil && len(*probesConfig.FileDirectory) > 0 {
+		fileDirectory := strings.TrimRight(*probesConfig.FileDirectory, "/")
+		if len(fileDirectory) > 0 {
+			cmdList = append(cmdList, fmt.Sprintf("-f %s/%s", fileDirectory, probeFile))
+		}
+	}
+	probe.Exec = &corev1.ExecAction{
+		Command: []string{"/bin/sh", "-c", strings.Join(cmdList, FlagDelimiterSpace)},
+	}
+}
+
+func getProbeWithoutHandlers(probe *corev1.Probe) *corev1.Probe {
+	probe = getOrInitProbe(probe)
+	probe.ProbeHandler = corev1.ProbeHandler{}
+	return probe
+}
+
+func getOrInitProbe(probe *corev1.Probe) *corev1.Probe {
+	if probe == nil {
+		return &corev1.Probe{}
+	}
+	return probe
+}
+
+func patchFileBasedProbe(instance *olv1.OpenLibertyApplication, defaultProbe *corev1.Probe, instanceProbe *corev1.Probe, scriptName string, probeFile string) *corev1.Probe {
+	defaultProbe = getOrInitProbe(defaultProbe)
+	instanceProbe = getOrInitProbe(instanceProbe)
+	isExecConfigured := instanceProbe.Exec != nil
+	instanceProbe = rcoutils.CustomizeProbeDefaults(instanceProbe, defaultProbe)
+	if !isExecConfigured {
+		configureFileBasedProbeExec(instance, instanceProbe, scriptName, probeFile)
+	}
+	return instanceProbe
+}
+
+func CustomizeFileBasedProbes(pts *corev1.PodTemplateSpec, instance *olv1.OpenLibertyApplication) {
+	if !IsFileBasedProbesEnabled(instance) {
+		if instance.Spec.Probes == nil {
+			return
+		}
+		// Reset probe if file-based settings were previously configured
+		instance.Spec.Probes.Startup = clearFileBasedProbe(instance.Spec.Probes.Startup)
+		instance.Spec.Probes.Liveness = clearFileBasedProbe(instance.Spec.Probes.Liveness)
+		instance.Spec.Probes.Readiness = clearFileBasedProbe(instance.Spec.Probes.Readiness)
+		return
+	}
+	instance.Spec.Probes.Startup = patchFileBasedProbe(instance, instance.Spec.Probes.OpenLibertyApplicationProbes.GetDefaultStartupProbe(instance), instance.Spec.Probes.Startup, StartupProbeFileBasedScriptName, StartupProbeFileName)
```

Comment:
We should set or unset the probes directly on the PodTemplateSpec. We shouldn't set the probes in the CR, as we could potentially override user's config / create back-and-forth update with ArgoCD managed CRs. Also, we wouldn't be able to know whether users set a config or we did in the future.